### DISCUSSION TITLE
its possible to refactoring the display error

### DISCUSSION THREAD

**Discussion by p0W3RH311** [PowerShell/PowerShell#15660] [Enhancements]
hi guys !!

error message in **PS** is great and its inspired from **Rust** display error message. the team worked hard to implement this ... thanks for them and thanks for all the contributors. but there is a small problem, Rust doesn't have the best error message in the world. there is the **Elm** language it is he who has _**THE BEST DISPLAY ERROR**_ messages in the world.

the characteristics of **ELM** error messages are that they are user-friendly, diversify, coloring, emphasis..etc  in addition directs the coder to the right path and gives him suggestions.
I know it's hard to rewrite the error code again, but is it possible to take inspiration from **ELM** to make **PS
![Screenshot_2021-03-21 Core Language · An Introduction to Elm(1)](https://user-images.githubusercontent.com/36144546/111919502-914d3780-8a47-11eb-85ab-e0d3c2a8cc01.png)
![Screenshot_2021-03-21 Core Language · An Introduction to Elm](https://user-images.githubusercontent.com/36144546/111919506-95795500-8a47-11eb-9490-319f01111990.png)

** more beautiful

**Comment by SteveL-MSFT** [comment]
`$ErrorView` is an enum, so we can certainly extend it to support a new view.  The example you show would certainly be more useful to newer users.  The challenge I see is how many cases would need to be handled.  There is already fuzzing matching code for commands, so I can see this applied to cmdlets, parameters, parameter values, object methods/properties, .NET method/properties.  Not sure if we'd want to search all of .NET for types though.  What else?

The `$ErrorView` code is all PowerShell script, so if someone could prototype an example, we could use that for a new view.  Would also need a name for this new view.
